packages/flutter/test/material/navigation_bar_test.dart

```diff
@@ -425,6 +425,71 @@ void main() {
     }
   });
 
+    testWidgets('Material3 - NavigationBar label can scale and has maxScaleFactor', (WidgetTester tester) async {
+    const String label = 'A';
+
+    Widget buildApp({ required TextScaler textScaler }) {
+      return MediaQuery(
+        data: MediaQueryData(textScaler: textScaler),
+        child: Localizations(
+          locale: const Locale('en', 'US'),
+          delegates: const <LocalizationsDelegate<dynamic>>[
+            DefaultMaterialLocalizations.delegate,
+            DefaultWidgetsLocalizations.delegate,
+          ],
+          child: MaterialApp(
+            theme: ThemeData(useMaterial3: true),
+            home: Navigator(
+              onGenerateRoute: (RouteSettings settings) {
+                return MaterialPageRoute<void>(
+                  builder: (BuildContext context) {
+                    return Scaffold(
+                      bottomNavigationBar: NavigationBar(
+                        destinations: const <NavigationDestination>[
+                          NavigationDestination(
+                            label: label,
+                            icon: Icon(Icons.ac_unit),
+                          ),
+                          NavigationDestination(
+                            label: 'B',
+                            icon: Icon(Icons.battery_alert),
+                          ),
+                        ],
+                      ),
+                    );
+                  },
+                );
+              },
+            ),
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(buildApp(textScaler: TextScaler.noScaling));
+    expect(find.text(label), findsOneWidget);
+
+    if (!kIsWeb || isCanvasKit) { // https://github.com/flutter/flutter/issues/99933
+      expect(tester.getSize(find.text(label)), const Size(12.5, 16.0));
+    }
+
+    await tester.pumpWidget(buildApp(textScaler: const TextScaler.linear(1.1)));
+    await tester.pumpAndSettle();
+    if (!kIsWeb || isCanvasKit) {
```

Comment:
maybe we should also mention the issue comment(`// https://github.com/flutter/flutter/issues/99933`) here and below? Or maybe we can just skip `kIsWeb || !isCanvasKit` and label that with the issue comment like:
![Screenshot 2024-01-23 at 3 16 24 PM](https://github.com/flutter/flutter/assets/36861262/6c511375-006f-4f28-90a4-1b94f504ca5d)